appium-tests/ios/ios.spec.js

```diff
@@ -0,0 +1,352 @@
+                    compare: function (actual, expected) {
+                        var result = {
+                            pass: true,
+                            message: ''
+                        };
+                        if (actual.indexOf(expected) < 0)  {
+                            result.pass = false;
+                            result.message = 'Expected ' + actual + ' to contain ' + expected;
+                        }
+                        return result;
+                    }
+                };
+            }
+        });
+    });
+
+    it('camera.ui.util configuring driver and starting a session', function (done) {
+        stopFlag = true; // just in case of timeout
+        getDriver().then(function () {
+            stopFlag = false;
+        }, function (error) {
+            return fail(error);
+        })
+        .then(function() {
+            return driver.contexts();
+        })
+        .finally(done);
+    }, 5 * MINUTE);
+
+    it('camera.ui.util determine webview context name', function (done) {
+        return driver
+            .contexts()
+            .then(function onSuccess (contexts) {
+                var found = false;
+                var i;
+                // take the last named context
+                for (i = 0; i < contexts.length; i++) {
+                    if (contexts[i].indexOf('mobilespec') >= 0) {
+                        webviewContext = contexts[i];
+                        found = true;
+                    }
+                }
+                // if no named context is found then take the last webview context
+                if (!found) {
+                    for (i = 0; i < contexts.length ; i++) {
+                        if (contexts[i].indexOf('WEBVIEW') >= 0) {
+                            webviewContext = contexts[i];
+                        }
+                    }
+                }
+                console.log('Webview context is ' + webviewContext);
+                done();
+            }, function onError (err) {
+                console.log(err);
+            })
+            .finally(done);
+    }, MINUTE);
+
+    describe('specs', function () {
+        beforeEach(function (done) {
+            // prepare the app for the test
+            if (!stopFlag) {
+                return driver
+                    .context(webviewContext)
+                    .then(function () {
+                        return driver; // no-op
+                    }, function (error) {
+                        expect(true).toFailWithMessage(error);
+                    })
+                    .execute('document.getElementById("info").innerHTML = "' + STARTING_MESSAGE + '";')
+                    .finally(done);
+            }
+            done();
+        }, 3 * MINUTE);
+
+        afterEach(function (done) {
+            if (!errorFlag || stopFlag) {
+                // either there's no error or we've failed irrecoverably
+                // nothing to worry about!
```

Comment:
At this point, the failure already has been loudly announced, this logic is just to skip current beforeEach function.